lib/wasmex.ex

```diff
@@ -92,6 +92,29 @@ defmodule Wasmex do
   - `:f32` a 32 bit float
   - `:f64` a 64 bit float
 
+  ### Linking
+
+  Wasm links may be given as an additional option.
+  Links is an array of module name-bytes pair.
+
+      iex> wasm = File.read!(TestHelper.wasm_link_test_file_path())
+      iex> linked_wasm = File.read!(TestHelper.wasm_test_file_path())
+      iex> links = [%{name: "utils", bytes: linked_wasm}]
+      iex> {:ok, pid} = Wasmex.start_link(%{bytes: wasm, links: links})
+      iex> Wasmex.call_function(pid, "sum_range", [1, 5])
+      {:ok, [15]}
+
+  It is also possible to link a module that was compiled for improved performance:
+
+      iex> wasm = File.read!(TestHelper.wasm_link_test_file_path())
+      iex> linked_wasm = File.read!(TestHelper.wasm_test_file_path())
+      iex> {:ok, store} = Wasmex.Store.new()
+      iex> {:ok, compiled_module} = Wasmex.Module.compile(store, linked_wasm)
+      iex> links = [%{name: "utils", module: compiled_module}]
+      iex> {:ok, pid} = Wasmex.start_link(%{bytes: wasm, links: links, store: store})
+      iex> Wasmex.call_function(pid, "sum_range", [1, 5])
+      {:ok, [15]}
```

Comment:
@tessi good callout on changing from `list` to `map` I think it makes a ton of sense and would prevent name-conflicts outright.

I am more inclined towards a map-of-map so that we don't have to introduce an API change in the future when we want to pass in additional options per link.